api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java

```diff
@@ -328,7 +328,15 @@ public String getReadOnlyHtml(
             .getBucketName();
 
     Blob blob = getBlobWithSizeConstraint(bucketName, notebookName);
-    return convertNotebookToHtml(blob.getContent());
+
+    if (NotebookUtils.isJupyterNotebook(notebookName)) {
+      return convertJupyterNotebookToHtml(blob.getContent());
+    } else if (NotebookUtils.isRstudioNotebook(notebookName)) {
+      return convertRstudioNotebookToHtml(blob.getContent());
+    } else {
+      throw new NotImplementedException(
+          String.format("%s type of file is not implemented yet", notebookName));
```

Comment:
Let's add a little more context here: "Converting %s to read-only HTML is not supported" or similar